### PR TITLE
fix(inlineerror): adding inline error on empty state

### PR DESCRIPTION
**Pull request by il10**
Refactored ActiveState and ActivityFeed components to bring up deeplink/inlineerror logic to ActivityFeed. ActiveState (now named ActivityFeedItem) now only controls rendering the feed items.

Added inline error logic for when deeplinked item is deleted and shows the empty state.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @il10 has signed the CLA. Thanks for the pull request!

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeedItem.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import AppActivity from '../app-activity';
+import Comment from '../comment';
+import TaskNew from '../task-new';
+import Version, { CollapsedVersion } from '../version';
+import Keywords from '../keywords';
+import withErrorHandling from '../../withErrorHandling';
+import type { FocusableFeedItemType } from '../../../../common/types/feed';
+
+type Props = {
+    action: string | 'applied',
```
Comment (mxiao6) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeedItem.js]: This `action` prop is required here but I didn't see it in the `ActivityFeed` when you use `ActivityFeedItem`.

File: flow-typed/box-ui-elements-updates.js [hunk 2]
```diff
@@ -0,0 +1,931 @@
+/**
```
Comment (ConradJChan) [inline flow-typed/box-ui-elements-updates.js]: What is this file used for?

File: src/elements/content-sidebar/activity-feed/activity-feed/__tests__/ActivityFeedItem-test.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+const getWrapper = (params = {}) => mount(<ActivityFeedItem currentUser={currentUser} {...params} />);
+
+describe('elements/content-sidebar/ActivityFeedItem/activity-feed/ActivityFeedItem', () => {
+    test.each`
+        item                  | expected
+        ${comment}            | ${'[data-testid="comment"]'}
+        ${taskWithAssignment} | ${'[data-testid="task"]'}
+        ${fileVersion}        | ${'[data-testid="version"]'}
+        ${appActivity}        | ${'[data-testid="app-activity"]'}
```
Comment (mxiao6) [inline src/elements/content-sidebar/activity-feed/activity-feed/__tests__/ActivityFeedItem-test.js]: Should we also test case `keywords` here?

File: flow-typed/box-ui-elements.js [hunk 4]
```diff
@@ -732,16 +741,22 @@ type AppActivityAPIItems = {
 };
 
 type AppActivityItem = {
+    activity_template: ActivityTemplateItem,
+    app: AppItem,
     created_at: string,
+    created_by: User,
+    id: string,
     permissions: BoxItemPermission,
-} & BaseAppActivityItem;
```
Comment (ConradJChan) [inline flow-typed/box-ui-elements.js]: What is the intention of this removal?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js [hunk 5]
```diff
@@ -216,36 +245,48 @@ class ActivityFeed extends React.Component<Props, State> {
                             <LoadingIndicator />
                         </div>
                     )}
+                    {!isLoading && (
+                        <ul className="bcs-activity-feed-active-state">
```
Comment (ConradJChan) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js]: Should we update the classname?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js [hunk 6]
```diff
@@ -216,36 +245,48 @@ class ActivityFeed extends React.Component<Props, State> {
...
+                            {isEmpty && <EmptyState showCommentMessage={showCommentForm} />}
+                            {!isEmpty &&
+                                collapseFeedState(feedItems).map((item: FeedItem) => {
```
Comment (ConradJChan) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js]: Can we call `collapseFeedState` once and use it both here and L231?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js [hunk 7]
```diff
@@ -6,15 +6,27 @@
...
 import type { FocusableFeedItemType } from '../../../../common/types/feed';
 import './ActivityFeed.scss';
+import messages from './messages';
+
+const errorMessageByEntryType = {
+    comment: messages.commentMissingError,
+    task: messages.taskMissingError,
+};
+
+const hasErrorMessageForEntryType = feedEntryType => {
```
[the remark this answers is not in the document]
Comment (il10) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js]: I think having this function makes the code more readable for when I use it within the inline error func

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js [hunk 8]
```diff
@@ -216,36 +245,48 @@ class ActivityFeed extends React.Component<Props, State> {
...
-                            mentionSelectorContacts={mentionSelectorContacts}
-                            getMentionWithQuery={getMentionWithQuery}
-                            approverSelectorContacts={approverSelectorContacts}
-                            getApproverWithQuery={getApproverWithQuery}
-                            activeFeedEntryId={activeFeedEntryId}
-                            activeFeedEntryType={activeFeedEntryType}
-                            activeFeedItemRef={this.activeFeedItemRef}
-                        />
+                                    return (
+                                        <ActivityFeedItem
```
Comment (ConradJChan) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js]: I'm not sure that I love how `ActivityFeedItem` has to take in all the possible props for the various feed item types, somehow it felt more reasonable for `ActiveState`

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js [hunk 9]
```diff
@@ -216,36 +245,48 @@ class ActivityFeed extends React.Component<Props, State> {
...
+                                            onVersionInfo={onVersionHistoryClick ? this.openVersionHistoryPopup : null}
+                                            translations={translations}
+                                            getAvatarUrl={getAvatarUrl}
+                                            getUserProfileUrl={getUserProfileUrl}
+                                            mentionSelectorContacts={mentionSelectorContacts}
+                                            getMentionWithQuery={getMentionWithQuery}
+                                            approverSelectorContacts={approverSelectorContacts}
+                                            getApproverWithQuery={getApproverWithQuery}
+                                            activeFeedEntryId={activeFeedEntryId}
+                                            activeFeedEntryType={activeFeedEntryType}
```
Comment (ConradJChan) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js]: It doesn't seem like these are needed props on `ActivityFeedItem`?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeedItem.js [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+    getAvatarUrl,
+    getUserProfileUrl,
+    isFocused,
+    activeFeedItemRef,
+}: Props): React.Node => {
+    if (!item) {
+        return null;
+    }
+    const { type, id, permissions } = item;
+    if (!currentUser) return null;
```
Comment (ConradJChan) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeedItem.js]: This wasn't the previous behavior of `ActiveState`, is this desired?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (mxiao6) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js]: Ok, you can also simplify it to something like `const hasErrorMessageForEntryType = feedEntryType => !!errorMessageByEntryType[feedEntryType];`

File: flow-typed/box-ui-elements-updates.js (hunk 2 above)
Comment (il10) [reply flow-typed/box-ui-elements-updates.js]: I had copied over some changes I made in box-ui-elements.js for my own reference. Deleted it once I pushed to make the PR

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js (hunk 8 above)
Comment (jstoffan) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js]: I agree, I don't think this is the direction we want to go. If we want to refactor here, it seems like each type of activity item should be moved to its own component first. We should avoid creating more mega-components.

**Comment by mickr** [comment]
Has there been any movement on this PR?

**Pull request closed without merging**